enhancements/network/network-policy-audit-logging.md

```diff
@@ -0,0 +1,315 @@
+- Allow the Customer to control the logging rate for this feature via the Cluster Network Operator's config 
+- Allow the Customer to configure this fetaure's use and level via a Namespace's `.metadata.annotations` field 
+- Collect the relevant data from the ovn-controller logs and preset it to the Customer in an seamless manner 
+
+### Non-Goals
+
+- Network Policy Object Logging (`oc describe <entitiy>` takes care of that)
+
+## Proposal
+
+The Enhancment's implementation starts with somce changes to the [network.operator.openshift.io](https://github.com/openshift/api/blob/master/operator/v1/types_network.go) API.  These changes involve adding one optional flag to the `OVNKubernetesConfig` struct as follows. 
+
+```go
+// ovnKubernetesConfig contains the configuration parameters for networks
+// using the ovn-kubernetes network project
+type OVNKubernetesConfig struct {
+	// mtu is the MTU to use for the tunnel interface. This must be 100
+	// bytes smaller than the uplink mtu.
+	// Default is 1400
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	MTU *uint32 `json:"mtu,omitempty"`
+	// geneve port is the UDP port to be used by geneve encapulation.
+	// Default is 6081
+	// +kubebuilder:validation:Minimum=1
+	// +optional
+	GenevePort *uint32 `json:"genevePort,omitempty"`
+	// HybridOverlayConfig configures an additional overlay network for peers that are
+	// not using OVN.
+	// +optional
+	HybridOverlayConfig *HybridOverlayConfig `json:"hybridOverlayConfig,omitempty"`
+	// ipsecConfig enables and configures IPsec for pods on the pod network within the
+	// cluster.
+	// +optional
+	IPsecConfig *IPsecConfig `json:"ipsecConfig,omitempty"`
+  // aclLoggingRateLimit allows for the configuring of max acl logging rate 
+  // +optional
+  aclLoggingRateLimit  *uint32 `json:"aclLoggingRateLimit,omitempty"`
+
+```
+
+The flag `aclLoggingRateLimit` defaults to `20`. 
+
+This will result in an updated OVNKubernetes Operator configuration object: (.spec.defaultNetwork.ovnKubernetesConfig) which will enable cluster-wide configuration of acl logging at cluster installation time: 
+
+```go
+spec:
+  defaultNetwork:
+    type: OVNKubernetes
+    ovnKubernetesConfig:
+      mtu: 1400
+      genevePort: 6081
+      ipsecConfig: {}
+      aclLoggingRateLimit: 20
+```
+
+### User Stories
+
+Detail the things that people will be able to do if this is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+
+Include a story on how this proposal will be operationalized:  lifecycled, monitored and remediated at scale.
+
+#### Story 1
+
+As an Openshift user I want to see what Network Policies and accompanied ACLs are dropping the most traffic in a cluster. For example, when a packet runs into a `drop` ACL it will log a message similar to the following 
```

Comment:
- Should be consistent with Openshift vs OpenShift and Network Policies vs NetworkPolicies (global comment).
- End the last sentence with a colon.